### PR TITLE
Implement concurrent config connections limit for native server

### PR DESCRIPTION
**Pull request by NikolaiShipilov**
## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] Pull request title reflects its content
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Description:
* Add server config property MaxAllowedConfigConnections to limit count of concurrent config connections
* Instantiate the config server only for native connections which use the config protocol
* Introduce new config packet type PacketType::ConnectionRejected = 0x86
* Fix thread joining issue when the native device connection failed with an exception

### CODE REVIEW

File: shared/libraries/native_streaming_protocol/src/native_streaming_server_handler.cpp [hunk 1]
```diff
@@ -353,6 +366,38 @@ void NativeStreamingServerHandler::setUpConfigProtocolCallbacks(std::shared_ptr<
...
+        std::scoped_lock lock(sync);
+
+        // Rejects new conenctions in case if count of concurrent config connections exceed the limit
+        // defined with "MaxAllowedConfigConnections" property (where 0 value stands for unlimited count)
+        if (maxAllowedConfigConnections != 0 && configConnectionsCount == maxAllowedConfigConnections)
+        {
+            auto reply = config_protocol::ConfigProtocolServer::generateConnectionRejectedReply(
+                firstPacketBuffer.getId(),
+                OPENDAQ_ERR_CONNECTION_LIMIT_REACHED,
+                "Native configuration connection rejected - connections limit reached on server",
```
Comment (dejancrn) [inline shared/libraries/native_streaming_protocol/src/native_streaming_server_handler.cpp:388]: Usually, the max native configuration connections are either 0 or 1. Is it possible to inject the IP(s) of the already connected clients into the error message?

**Review by dejancrn — APPROVED** [review]

File: shared/libraries/native_streaming_protocol/src/native_streaming_server_handler.cpp (hunk 1 above)
Comment (NikolaiShipilov) [reply shared/libraries/native_streaming_protocol/src/native_streaming_server_handler.cpp:388]: The IP address of the connected client is currently not accessible to the higher level code, but the native library interface could be extended to enable this.

**Review by janm-dw — APPROVED** [review]

File: shared/libraries/native_streaming_protocol/src/native_streaming_server_handler.cpp (hunk 1 above)
Comment (JakaMohorkoDS) [reply shared/libraries/native_streaming_protocol/src/native_streaming_server_handler.cpp:388]: This would be great to add, as it eases debugging a lot.

**Review by JakaMohorkoDS — APPROVED** [review]
Lgtm. As Dejan mentioned, it'd be great to be able to receive the IP of the currently connected clients when the connection is refused.

**Pull request merged**